### PR TITLE
feat(core): Allow to pass validationRules to shop and admin GraphQL API

### PR DESCRIPTION
**Pull request by hendrik-advantitge**
Our goal with this PR is to use packages like [https://www.npmjs.com/package/graphql-depth-limit](url), to prevent large queries from bringing down the server.

### CODE REVIEW

**Comment by michaelbromley** [comment]
Hi, thanks for the PR. This looks useful! Could you make the PR against the `master` branch instead?

**Comment by hendrik-advantitge** [comment]
Done!

File: packages/core/src/config/vendure-config.ts [hunk 1]
```diff
@@ -106,6 +107,22 @@ export interface ApiOptions {
...
+     * Custom functions to use as additional validation rules when validating the schema for the admin GraphQL API
+     * [ApolloServer playground](https://www.apollographql.com/docs/apollo-server/api/apollo-server/#constructoroptions-apolloserver).
+     *
+     * @default []
+     */
+    adminApiValidationRules?: ((context: ValidationContext) => any)[];
+    /**
+     * @description
+     * Custom functions to use as additional validation rules when validating the schema for the shop GraphQL API
+     * [ApolloServer playground](https://www.apollographql.com/docs/apollo-server/api/apollo-server/#constructoroptions-apolloserver).
```
Comment (michaelbromley) [inline packages/core/src/config/vendure-config.ts]: These links seem to be mis-labeled, and is it supposed to link to https://www.apollographql.com/docs/apollo-server/api/apollo-server/#validationrules?

Comment (hendrik-advantitge) [reply packages/core/src/config/vendure-config.ts]: Good catch, should be fixed now

**Pull request merged**